drivers/sensor/iis2iclx/iis2iclx_i2c.c

```diff
@@ -46,7 +47,7 @@ int iis2iclx_i2c_init(const struct device *dev)
 	data->ctx_i2c.write_reg = (stmdev_write_ptr) iis2iclx_i2c_write,
 
 	data->ctx = &data->ctx_i2c;
-	data->ctx->handle = data;
+	data->ctx->handle = (struct device *)dev;
```

Comment:
Casting away `const` is illegal.  I'm not clear on how `stmdev_ctx_t` uses this value, but if possible it would be better to keep this as `data` so `data->dev` (or some other field that's a back-pointer) can be used to access the device pointer legally.

Reply:
@pabigot
Sure! Moreover we already have a back pointer to `struct device` inside data.
I do the modification and re-test.